web/src/engine/websites/CarToonMad.ts

```diff
@@ -0,0 +1,34 @@
+import { Tags } from '../Tags';
+import icon from './CarToonMad.webp';
+import { type Chapter, DecoratableMangaScraper, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchRequest } from '../FetchProvider';
+
+@Common.MangaCSS(/^https?:\/\/(www\.)?cartoonmad\.com\/[^/]+/, 'td:nth-child(2)  tr:nth-child(3) > td:nth-child(2) > a:last-child')
+@Common.MangasMultiPageCSS('/comic99.0{page}.html', 'a.a1')
+@Common.ChaptersSinglePageCSS('#info td > a')
+@Common.ImageDirect()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('cartoonmad', `CarToonMad`, 'https://www.cartoonmad.com', Tags.Language.Chinese, Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const referer = new URL(chapter.Identifier, this.URI);

```

Comment:
May use `referer = new URL(...).href;` since only `href` is used in this function